api/src/controllers/internal/processProgram.ts

```diff
@@ -47,6 +52,42 @@ export const processProgram = async (
       // copy the code.js program to log and end write stream
       writeStream.end(program)
 
+      session.completed = true
+      console.log('session completed', session)
+    } catch (err: any) {
+      session.completed = true
+      session.crashed = err.toString()
+      console.log('session crashed', session.id, session.crashed)
```

Comment:
is this `console.log` left for a reason?

Reply:
Yes, we are intentionally using it here to log the status of the session. In the future, we will use some other logger then we'll replace it with that.